src/test/groovy/graphql/schema/PropertyDataFetcherTest.groovy

```diff
@@ -458,5 +458,12 @@ class PropertyDataFetcherTest extends Specification {
 
         then:
         result == "bar"
+
+        // repeat - should be cached
+        when:
+        result = propertyDataFetcher.get(dfe)
+
+        then:
+        result == "bar"
     }
```

Comment:
the tests all pass and I tweaked this one to ensure out caching works for `Foo extends Bar` case